Build failure: text-engine

## Steps To Reproduce

Steps to reproduce the behavior:

1. build text-engine

## Build log

nix log /nix/store/d3qiiv386z3nz425xg9z5f6r71nkba2n-text-engine-0.1.1.drv
<details>

<summary>Build Log</summary>

nix log /nix/store/d3qiiv386z3nz425xg9z5f6r71nkba2n-text-engine-0.1.1.drv | grep error
```
../src/layout/layout-box.c:114:62: error: passing argument 1 of 'g_type_name_from_instance' from incompatible pointer type [-Wincompatible-pointer-types]
../src/layout/layout.c:99:68: error: passing argument 1 of 'g_type_name_from_instance' from incompatible pointer type [-Wincompatible-pointer-types]
../src/layout/layout.c:110:69: error: passing argument 1 of 'g_type_name_from_instance' from incompatible pointer type [-Wincompatible-pointer-types]
../src/layout/layout.c:120:60: error: passing argument 1 of 'g_type_name_from_instance' from incompatible pointer type [-Wincompatible-pointer-types]
../src/ui/display.c:123:44: error: passing argument 2 of 'layout_snapshot_recursive' from incompatible pointer type [-Wincompatible-pointer-types]

```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

## Notify maintainers

@foo-dogsquared 

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

The build also fails for me:

```
error: builder for '/nix/store/m37abdhvgg27qd70nifhxlp5mw3jrizg-text-engine-0.1.1.drv' failed with exit code 1;
       last 25 log lines:
       > ../demo/demo.c:77:17: warning: unused variable 'self' [-Wunused-variable]
       >    77 |     DemoWindow *self = DEMO_WINDOW (object);
       >       |                 ^~~~
       > ../demo/demo.c:74:41: warning: unused parameter 'value' [-Wunused-parameter]
       >    74 |                           const GValue *value,
       >       |                           ~~~~~~~~~~~~~~^~~~~
       > ../demo/demo.c: In function 'main':
       > ../demo/demo.c:155:5: warning: 'G_APPLICATION_FLAGS_NONE' is deprecated: Use 'G_APPLICATION_DEFAULT_FLAGS' instead [-Wdeprecated-declarations]
       >   155 |     app = adw_application_new ("com.mattjakeman.TextEngine.Demo", G_APPLICATION_FLAGS_NONE);
       >       |     ^~~
       > In file included from /nix/store/ck6zfmyhb81wjq86zy6jxdisf3i89a5x-glib-2.82.4-dev/include/glib-2.0/gio/giotypes.h:30,
       >                  from /nix/store/ck6zfmyhb81wjq86zy6jxdisf3i89a5x-glib-2.82.4-dev/include/glib-2.0/gio/gio.h:28,
       >                  from /nix/store/j073nmj1bbfpk50vh0sk256za3iqpxrz-gtk4-4.16.12-dev/include/gtk-4.0/gtk/css/gtkcsssection.h:24,
       >                  from /nix/store/j073nmj1bbfpk50vh0sk256za3iqpxrz-gtk4-4.16.12-dev/include/gtk-4.0/gtk/css/gtkcss.h:36,
       >                  from /nix/store/j073nmj1bbfpk50vh0sk256za3iqpxrz-gtk4-4.16.12-dev/include/gtk-4.0/gtk/gtk.h:29,
       >                  from /nix/store/1b81027id6mi675vpkdidk4066j0z1bb-libadwaita-1.6.3-dev/include/libadwaita-1/adwaita.h:9,
       >                  from ../demo/demo.c:26:
       > /nix/store/ck6zfmyhb81wjq86zy6jxdisf3i89a5x-glib-2.82.4-dev/include/glib-2.0/gio/gioenums.h:1545:3: note: declared here
       >  1545 |   G_APPLICATION_FLAGS_NONE GIO_DEPRECATED_ENUMERATOR_IN_2_74_FOR(G_APPLICATION_DEFAULT_FLAGS),
       >       |   ^~~~~~~~~~~~~~~~~~~~~~~~
       > ../demo/demo.c: At top level:
       > ../demo/demo.c:46:20: warning: 'properties' defined but not used [-Wunused-variable]
       >    46 | static GParamSpec *properties [N_PROPS];
       >       |                    ^~~~~~~~~~
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/m37abdhvgg27qd70nifhxlp5mw3jrizg-text-engine-0.1.1.drv'.
```

The full logs: [log.txt](https://github.com/user-attachments/files/18724489/log.txt)

--- a/pkgs/by-name/te/text-engine/package.nix
+++ b/pkgs/by-name/te/text-engine/package.nix
@@ -17,21 +17,14 @@
 
 stdenv.mkDerivation rec {
   pname = "text-engine";
-  version = "0.1.1";
+  version = "0.1.1-unstable-2024-09-16";
+
   src = fetchFromGitHub {
     owner = "mjakeman";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-YSG4Vk3hrmtaJkK1WAlQcdgiDdgC4Un0t6UdaoIcUes=";
+    rev = "4c26887556fd8e28211324c4058d49508eb5f557";
+    hash = "sha256-0rMBz2s3wYv7gZiJTj8rixWxBjT6Dd6SaINP8kDbTyw=";
   };
-
-  patches = [
-    # Fixes build with newer versions of clang
-    (fetchpatch {
-      url = "https://github.com/mjakeman/text-engine/commit/749c94d853c0b0e29e79a1b270ec61947b65c319.patch";
-      hash = "sha256-vs/a8IBovArw8tc1ZLUsaDHRVyA71KMB1NGENOKNOdk=";
-    })
-  ];
 
   nativeBuildInputs = [
     gobject-introspection

--- a/pkgs/by-name/te/text-engine/package.nix
+++ b/pkgs/by-name/te/text-engine/package.nix
@@ -1,16 +1,18 @@
 {
-  stdenv,
   lib,
+  stdenv,
   fetchFromGitHub,
   fetchpatch,
+
+  gobject-introspection,
+  gtk4,
   meson,
   ninja,
-  json-glib,
-  gtk4,
-  libxml2,
-  gobject-introspection,
   pkg-config,
+
+  json-glib,
   libadwaita,
+  libxml2,
 }:
 
 stdenv.mkDerivation rec {
@@ -45,14 +47,14 @@ stdenv.mkDerivation rec {
     libxml2
   ];
 
-  meta = with lib; {
+  meta = {
     description = "Rich text framework for GTK";
     mainProgram = "text-engine-demo";
     homepage = "https://github.com/mjakeman/text-engine";
-    license = with licenses; [
+    license = with lib.licenses; [
       mpl20
       lgpl21Plus
     ];
-    maintainers = with maintainers; [ foo-dogsquared ];
+    maintainers = with lib.maintainers; [ foo-dogsquared ];
   };
 }

--- a/pkgs/by-name/te/text-engine/package.nix
+++ b/pkgs/by-name/te/text-engine/package.nix
@@ -36,9 +36,14 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     libadwaita
-    json-glib
     libxml2
   ];
+
+  postPatch = ''
+    # See https://github.com/mjakeman/text-engine/pull/42
+    substituteInPlace src/meson.build \
+      --replace-fail "dependency('json-glib-1.0')," ""
+  '';
 
   meta = {
     description = "Rich text framework for GTK";

